tests/drivers/timer/nrf_grtc_timer/src/main.c

```diff
@@ -37,4 +49,69 @@ ZTEST(nrf_grtc_timer, test_get_ticks)
 		     "Unexpected result %" PRId64 " (expected: %" PRId64 ")", ticks, exp_ticks);
 }
 
+ZTEST(nrf_grtc_timer, test_timer_count_in_compare_mode)
+{
+	int err;
+	int key;
+	uint64_t test_ticks = 0;
+	char user_data[] = "test_timer_abort_in_compare_mode test\n";
+	int32_t channel = z_nrf_grtc_timer_chan_alloc();
+
+	TC_PRINT("Allocated GRTC channel %d\n", channel);
+	if (channel < 0) {
+		TC_PRINT("Failed to allocate GRTC channel, chan=%d\n", channel);
+		ztest_test_fail();
+	}
+
+	compare_isr_call_counter = 0;
+	key = z_nrf_grtc_timer_compare_int_lock(channel);
+	test_ticks = z_nrf_grtc_timer_read() + K_MSEC(TIMER_COUNT_TIME_MS).ticks;
+	err = z_nrf_grtc_timer_set(channel, test_ticks, timer_compare_interrupt_handler,
+				   (void *)user_data);
+	z_nrf_grtc_timer_compare_int_unlock(channel, key);
+	zassert_equal(z_nrf_grtc_timer_compare_read(channel), test_ticks,
+		      "Compare register set failed");
```

Comment:
`K_TICKS()` and `K_TIMEOUT_EQ()` should be used for this.

Reply:
Please provide an example i am now aware of how this macro works exactly.